### ISSUE TITLE
6.4 - TypeScript custom paths throwing ModuleNotFoundError in MDX

### ISSUE THREAD

**Issue by dsimon-onventis** [storybookjs/storybook#16711]
**Describe the bug**

In Storybook 6.3 with Angular, I was able to use the import `paths` defined in my `tsconfig.json` file, in my MDX stories. After updating to 6.4 RC, I am getting a build error when trying to import those paths.

**tsconfig:**
```
{
  "compilerOptions": {
    "paths": {
      "reusable-stuff": [
        "src/some-reusable-stuff.ts"
      ]
    }
  }
}
```

**MDX:**
```
import { myText } from 'reusable-stuff';
```

**To Reproduce**
https://github.com/dsimon-onventis/storybook-ts-paths-error-repro

On branch [`storybook-6.3`](https://github.com/dsimon-onventis/storybook-ts-paths-error-repro/tree/storybook-6.3), with Storybook 6.3 running Angular 12, the build succeeds.

**System**
Environment Info:

  System:                                          
    OS: Linux 5.10 Ubuntu 20.04.3 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i5-8365U CPU @ 1.60GHz
  Binaries:
    Node: 16.13.0 - /usr/local/bin/node
    Yarn: 1.22.15 - /usr/bin/yarn
    npm: 8.1.3 - ~/.npm-global/bin/npm
  Browsers:
    Chrome: 95.0.4638.69
  npmPackages:
    @storybook/addon-actions: ^6.4.0-rc.3 => 6.4.0-rc.3
    @storybook/addon-essentials: ^6.4.0-rc.3 => 6.4.0-rc.3
    @storybook/addon-links: ^6.4.0-rc.3 => 6.4.0-rc.3
    @storybook/angular: ^6.4.0-rc.3 => 6.4.0-rc.3
    @storybook/builder-webpack5: ^6.4.0-rc.3 => 6.4.0-rc.3
    @storybook/manager-webpack5: ^6.4.0-rc.3 => 6.4.0-rc.3

**Additional context**
<details>
  <summary>Error logs</summary>

ModuleNotFoundError: Module not found: Error: Can't resolve 'reusable-stuff' in '/home/dsimon/projects/storybook-ts-path-error-repro/src/stories'
    at /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/webpack/lib/Compilation.js:2004:28
    at /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/webpack/lib/NormalModuleFactory.js:795:13
    at eval (eval at create (/home/dsimon/projects/storybook-ts-path-error-repro/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:10:1)
    at /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/webpack/lib/NormalModuleFactory.js:275:22
    at eval (eval at create (/home/dsimon/projects/storybook-ts-path-error-repro/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/webpack/lib/NormalModuleFactory.js:431:22
    at /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/webpack/lib/NormalModuleFactory.js:124:11
    at /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/webpack/lib/NormalModuleFactory.js:667:25
    at /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/webpack/lib/NormalModuleFactory.js:852:8
    at /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/webpack/lib/NormalModuleFactory.js:972:5
    at /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/neo-async/async.js:6883:13
    at /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/webpack/lib/NormalModuleFactory.js:955:45
    at finishWithoutResolve (/home/dsimon/projects/storybook-ts-path-error-repro/node_modules/enhanced-resolve/lib/Resolver.js:296:11)
    at /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/enhanced-resolve/lib/Resolver.js:362:15
    at /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/enhanced-resolve/lib/Resolver.js:410:5
    at eval (eval at create (/home/dsimon/projects/storybook-ts-path-error-repro/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
resolve 'reusable-stuff' in '/home/dsimon/projects/storybook-ts-path-error-repro/src/stories'
  Parsed request is a module
  using description file: /home/dsimon/projects/storybook-ts-path-error-repro/package.json (relative path: ./src/stories)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      /home/dsimon/projects/storybook-ts-path-error-repro/src/stories/node_modules doesn't exist or is not a directory
      /home/dsimon/projects/storybook-ts-path-error-repro/src/node_modules doesn't exist or is not a directory
      looking for modules in /home/dsimon/projects/storybook-ts-path-error-repro/node_modules
        single file module
          using description file: /home/dsimon/projects/storybook-ts-path-error-repro/package.json (relative path: ./node_modules/reusable-stuff)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/reusable-stuff doesn't exist
            .mjs
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/reusable-stuff.mjs doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/reusable-stuff.js doesn't exist
            .jsx
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/reusable-stuff.jsx doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/reusable-stuff.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/reusable-stuff.tsx doesn't exist
            .json
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/reusable-stuff.json doesn't exist
            .cjs
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/reusable-stuff.cjs doesn't exist
        /home/dsimon/projects/storybook-ts-path-error-repro/node_modules/reusable-stuff doesn't exist
      /home/dsimon/projects/node_modules doesn't exist or is not a directory
      /home/dsimon/node_modules doesn't exist or is not a directory
      /home/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
      looking for modules in /home/dsimon/projects/storybook-ts-path-error-repro
        single file module
          using description file: /home/dsimon/projects/storybook-ts-path-error-repro/package.json (relative path: ./reusable-stuff)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/reusable-stuff doesn't exist
            .mjs
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/reusable-stuff.mjs doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/reusable-stuff.js doesn't exist
            .jsx
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/reusable-stuff.jsx doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/reusable-stuff.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/reusable-stuff.tsx doesn't exist
            .json
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/reusable-stuff.json doesn't exist
            .cjs
              Field 'browser' doesn't contain a valid alias configuration
              /home/dsimon/projects/storybook-ts-path-error-repro/reusable-stuff.cjs doesn't exist
        /home/dsimon/projects/storybook-ts-path-error-repro/reusable-stuff doesn't exist

</details>

**Comment by dsimon-onventis** [comment]
I've created branch [`storybook-6.3`](https://github.com/dsimon-onventis/storybook-ts-paths-error-repro/tree/storybook-6.3) in the reproduction repo, to proof that it worked with Storybook 6.3 running Angular 12.

**Comment by petterhoel** [comment]
I am seeing similar behaviour in nx repos where paths are defined in `<repo>/tsconfig.base.json`

Going from Storybook `6.3.0 => 6.4.0-rc.3`
and
nx `13.1.0 => 13.2.1`
angular `12 => 13`

here is my log output (some text redacted in <...>):
[stoybook-issue.txt](https://github.com/storybookjs/storybook/files/7582773/stoybook-issue.txt)

**Comment by petterhoel** [comment]
I can confirm same is happening in Windows 10
node `14.16.0`
npm `6.14.11`

To me this looks like a possible duplicate of #16759

**Comment by dsimon-onventis** [comment]
There is a workaround for this issue, it is described in #16759 and works well for me.

**Comment by dsimon-onventis** [comment]
Fixed with Storybook v6.4.5

**Comment by isqua** [comment]
Is it broken again in `@storybook/react@6.5.9`?
